pandas/tests/extension/sparse/test_sparse.py

```diff
@@ -0,0 +1,254 @@
+        # self.assert_series_equal(result, expected)
+
+
+class TestGetitem(base.BaseGetitemTests):
+
+    def test_get(self, data):
+        s = pd.Series(data, index=[2 * i for i in range(len(data))])
+        assert np.isnan(s.get(4)) and np.isnan(s.iloc[2])
+        assert s.get(2) == s.iloc[1]
+
+
+# Skipping TestSetitem, since we don't implement it.
+
+class TestMissing(base.BaseMissingTests):
+    @pytest.mark.skip(reason="Unsupported")
+    def test_fillna_limit_pad(self):
+        pass
+
+    @pytest.mark.skip(reason="Unsupported")
+    def test_fillna_limit_backfill(self):
+        pass
+
+    @pytest.mark.skip(reason="Unsupported")
+    def test_fillna_series_method(self):
+        pass
+
+    @pytest.mark.skip(reason="Unsupported")
+    def test_fillna_series(self):
+        # this one looks doable.
+        pass
+
+    def test_fillna_frame(self, data_missing):
+        # Have to override to specify that fill_value will change.
+        fill_value = data_missing[1]
+
+        result = pd.DataFrame({
+            "A": data_missing,
+            "B": [1, 2]
+        }).fillna(fill_value)
+
+        if pd.isna(data_missing.fill_value):
+            dtype = SparseDtype(data_missing.dtype, fill_value)
+        else:
+            dtype = data_missing.dtype
+
+        expected = pd.DataFrame({
+            "A": data_missing._from_sequence([fill_value, fill_value],
+                                             dtype=dtype),
+            "B": [1, 2],
+        })
+
+        self.assert_frame_equal(result, expected)
+
+
+class TestMethods(base.BaseMethodsTests):
+
+    def test_combine_le(self, data_repeated):
+        # We return a Series[SparseArray].__le__ returns a
+        # Series[Sparse[bool]]
+        # rather than Series[bool]
+        orig_data1, orig_data2 = data_repeated(2)
+        s1 = pd.Series(orig_data1)
+        s2 = pd.Series(orig_data2)
+        result = s1.combine(s2, lambda x1, x2: x1 <= x2)
+        expected = pd.Series(pd.SparseArray([
+            a <= b for (a, b) in
+            zip(list(orig_data1), list(orig_data2))
+        ], fill_value=False))
+        self.assert_series_equal(result, expected)
+
+        val = s1.iloc[0]
+        result = s1.combine(val, lambda x1, x2: x1 <= x2)
+        expected = pd.Series(pd.SparseArray([
+            a <= val for a in list(orig_data1)
+        ], fill_value=False))
+        self.assert_series_equal(result, expected)
+
+
+class TestCasting(base.BaseCastingTests):
+    pass
+
+
+class TestArithmeticOps(base.BaseArithmeticOpsTests):
+    series_scalar_exc = None
+    frame_scalar_exc = None
+    divmod_exc = None
+    series_array_exc = None
+
+    def test_error(self, data, all_arithmetic_operators):
+        # not sure
+        pass
+
+    @pytest.mark.xfail(reason="TODO", strict=True)
+    def test_divmod(self, data):
+        super().test_divmod(data)
+
+    @pytest.mark.xfail(reson="what is this test doing?", strict=True)
```

Comment:
I don't understand this base test. It generates a list of scalars at https://github.com/pandas-dev/pandas/blob/2be2ba570453ab09f0c8da46dfd8e1a179771fee/pandas/tests/extension/base/ops.py#L72 and then calls `Series.combine(data, list_of_scalars)`, which can't possibly do the right thing.